autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java

```diff
@@ -182,28 +192,41 @@ private void publish(TimeAlert alert) {
         event.setExistingClusterNodeCount(stackV4Response.getNodeCount());
         event.setScalingAdjustmentType(REGULAR);
         scalingActivityMsg = messagesService.getMessageWithArgs(AUTOSCALE_SCHEDULE_BASED_UPSCALE, targetIncrementNodeCount);
+
+        Set<String> servicesRunningOnHostgroup = stackV4Response.getServicesOnHostGroups().get(alert.getScalingPolicy().getHostGroup());
         if (targetIncrementNodeCount < 0) {
-            populateDecommissionCandidates(event, stackV4Response, alert.getCluster(), alert.getScalingPolicy(), -targetIncrementNodeCount);
+            populateDecommissionCandidates(event, stackV4Response, alert.getCluster(),
+                    alert.getScalingPolicy(), -targetIncrementNodeCount, servicesRunningOnHostgroup);
             status = SCHEDULE_BASED_DOWNSCALE;
             scalingActivityMsg = messagesService.getMessageWithArgs(AUTOSCALE_SCHEDULE_BASED_DOWNSCALE, event.getDecommissionNodeIds());
         }
         ScalingActivity scalingActivity = scalingActivityService.create(alert.getCluster(), status, scalingActivityMsg, clock.getCurrentTimeMillis());
         event.setActivityId(scalingActivity.getId());
+        if (servicesRunningOnHostgroup.contains(IMPALA)) {
+            if (!servicesRunningOnHostgroup.contains(YARN)) {
```

Comment:
There is not else so we can remove this if and put it as && condition for if above.

Reply:
Changed